docs/architecture/maui/mvvm.md

```diff
@@ -0,0 +1,364 @@
+- Always raising the PropertyChanged event at the end of the method that makes a property change, or when the object is known to be in a safe state. Raising the event interrupts the operation by invoking the event's handlers synchronously. If this happens in the middle of an operation, it might expose the object to callback functions when it is in an unsafe, partially updated state. In addition, it's possible for cascading changes to be triggered by PropertyChanged events. Cascading changes generally require updates to be complete before the cascading change is safe to execute.
+- Never raising a PropertyChanged event if the property does not change. This means that you must compare the old and new values before raising the PropertyChanged event.
+- Never raising the PropertyChanged event during a view model's constructor if you are initializing a property. Data-bound controls in the view will not have subscribed to receive change notifications at this point.
+- Never raising more than one PropertyChanged event with the same property name argument within a single synchronous invocation of a public method of a class. For example, given a NumberOfItems property whose backing store is the _numberOfItems field, if a method increments _numberOfItems fifty times during the execution of a loop, it should only raise property change notification on the NumberOfItems property once, after all the work is complete. For asynchronous methods, raise the PropertyChanged event for a given property name in each synchronous segment of an asynchronous continuation chain.
+
+A simple way to provide this functionality would be to create an extension of the BindableObject class. In this example, the ExtendedBindableObject class provides change notifications, which is shown in the following code example:
+
+```csharp
+public abstract class ExtendedBindableObject : BindableObject
+{
+    public void RaisePropertyChanged<T>(Expression<Func<T>> property)
+    {
+        var name = GetMemberInfo(property).Name;
+        OnPropertyChanged(name);
+    }
+
+    private MemberInfo GetMemberInfo(Expression expression)
+    {
+        ...
+    }
+}
+```
+
+.NET MAUI's BindableObject class implements the INotifyPropertyChanged interface, and provides an OnPropertyChanged method. The ExtendedBindableObject class provides the RaisePropertyChanged method to invoke property change notification, and in doing so uses the functionality provided by the BindableObject class.
+
+View model classes can then derive from the ExtendedBindableObject class. Therefore, each view model class uses the RaisePropertyChanged method in the ExtendedBindableObject class to provide property change notification. The following code example shows how the eShopOnContainers mobile app invokes property change notification by using a lambda expression:
+
+```csharp
+public bool IsLogin
+{
+    get
+    {
+        return _isLogin;
+    }
```

Comment:
```suggestion
    get => _isLogin;
```

Reply:
I will want to update the back-end code sample for this one to match, but that shouldn't take too long